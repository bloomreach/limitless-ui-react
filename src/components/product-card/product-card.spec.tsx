import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';

import { ProductCard } from './';

import Shirt from '../../../stories/assets/shirt.jpg';

const product = {
  id: '123',
  brand: 'Acme',
  title: 'Limitless Shirt',
  collection: 'Apparel',
  rating: 4.2,
  numRatings: 23,
  currency: 'USD',
  price: 25,
  salePrice: 20,
  image: Shirt,
  href: 'https://bloomreach.com',
};

describe('ProductCard', () => {
  it('should render successfully', () => {
    const { container: { firstElementChild } } = render(
      <ProductCard.Root>
        <ProductCard.Header>
          <img src={product.image} alt={product.title} />
        </ProductCard.Header>
        <ProductCard.Body>
          <ProductCard.Title>{product.title}</ProductCard.Title>
          <ProductCard.SubTitle>{product.collection}</ProductCard.SubTitle>
        </ProductCard.Body>
        <ProductCard.Footer>
        <ProductCard.Price currency={product.currency} price={product.price} />
          <ProductCard.Button variant="primary">Add to cart</ProductCard.Button>
        </ProductCard.Footer>
      </ProductCard.Root>
    );

    expect(firstElementChild).toMatchSnapshot();
  });
});
