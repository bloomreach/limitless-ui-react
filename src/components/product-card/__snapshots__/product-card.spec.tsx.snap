// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > should render successfully 1`] = `
<article
  class="lcui-product-card"
>
  <header
    class="lcui-product-card-header"
  >
    <img
      alt="Limitless Shirt"
      src="/stories/assets/shirt.jpg"
    />
  </header>
  <section
    class="lcui-product-card-main"
  >
    <div
      class="lcui-product-card-title"
    >
      Limitless Shirt
    </div>
    <div
      class="lcui-product-card-subtitle"
    >
      Apparel
    </div>
  </section>
  <footer
    class="lcui-product-card-footer"
  >
    <div
      class="lcui-product-card-price"
    >
      <span
        class="lcui-product-card-price-value"
      >
        $25.00
      </span>
    </div>
    <button
      class="lcui-product-card-btn lcui-product-card-btn--primary"
    >
      Add to cart
    </button>
  </footer>
</article>
`;
